test/shared/src/main/scala/zio/test/Assertion.scala

```diff
@@ -473,4 +473,52 @@ object Assertion {
         case t: Throwable => assertion(t)
       }
     }
+
+  final def isEmptyString: Assertion[String] = Assertion.assertion(s"isEmpty")() { actual =>
+    actual.isEmpty
+  }
+
+  final def isNonEmptyString: Assertion[String] = Assertion.assertion(s"isNonEmptyString")() { actual =>
+    actual.nonEmpty
+  }
+
+  final def equalsIgnoreCase(other: String): Assertion[String] = Assertion.assertion("equalsIgnoreCase")(param(other)) {
+    actual =>
+      actual.equalsIgnoreCase(other)
+  }
+
+  final def startsWith(prefix: String): Assertion[String] = Assertion.assertion("startsWith")(param(prefix)) { actual =>
+    actual.startsWith(prefix)
+  }
+
+  final def endsWith(suffix: String): Assertion[String] = Assertion.assertion("endsWith")(param(suffix)) { actual =>
+    actual.endsWith(suffix)
+  }
+
+  final def matchesRegex(regex: String): Assertion[String] = Assertion.assertion("matchesRegex")(param(regex)) {
+    actual =>
+      actual.matches(regex)
+  }
+
+  final def approximatelyEquals[A](range: NumericRange[A]): Assertion[A] =
```

Comment:
We generally try to avoid relying on implicit methods because they can be harder for the user to find. Can we change the signature to something like:

```scala
final def approximatelyEquals[A: Numeric](reference: A, tolerance: A): Assertion[A]
```